### PR TITLE
Osv: Add a fallback for the severity

### PR DESCRIPTION
**Pull request by fviernau**
Please see the individual commits.

Part of #3599.

### CODE REVIEW

File: advisor/src/main/kotlin/advisors/Osv.kt [hunk 1]
```diff
@@ -168,19 +171,26 @@ private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest? {
 }
 
 private fun Vulnerability.toOrtVulnerability(): org.ossreviewtoolkit.model.Vulnerability {
-    val (scoringSystem, severity) = severity.firstOrNull()?.let {
+    var (scoringSystem, score) = severity.firstOrNull()?.let {
```
Comment (sschuberth) [inline advisor/src/main/kotlin/advisors/Osv.kt:175]: IMO the variable should stay called "severity", esp. as that's also the field name of `databaseSpecific` you're reading out below.

File: clients/osv/src/main/kotlin/Model.kt [hunk 2]
```diff
@@ -25,6 +25,15 @@ import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.JsonObject
 
+// FIXME: Remove all JsonElement subtypes as property types from the model in favor of raw strings holding JSON.
```
Comment (sschuberth) [inline clients/osv/src/main/kotlin/Model.kt:28]: Can't you simply use a custom deserializer as outlined [here](https://github.com/Kotlin/kotlinx.serialization/issues/1058#issuecomment-983269638) that calls `encodeToString()` on the `JsonElement`? Performance should not be an issue here, as I do not expect these model element to be large.

File: advisor/src/main/kotlin/advisors/Osv.kt [hunk 3]
```diff
@@ -171,6 +171,7 @@ private fun createRequest(pkg: Package): VulnerabilitiesForPackageRequest? {
 }
 
 private fun Vulnerability.toOrtVulnerability(): org.ossreviewtoolkit.model.Vulnerability {
+    // FIXME: Clarify which entry is relevant and pick the right one accordingly.
```
Comment (sschuberth) [inline advisor/src/main/kotlin/advisors/Osv.kt:174]: So far, we basically only use `TODO` in the code base (modulo a few places that you introduced). As I don't see a relevant difference between `TODO` and `FIXME`, can we stich to `TODO` so we'd only need to search for one term instead of two?

File: advisor/src/funTest/kotlin/OsvFunTest.kt [hunk 4]
```diff
@@ -63,7 +63,8 @@ class OsvFunTest : StringSpec({
             .readValue<Map<Identifier, List<AdvisorResult>>>()
         val osv = createOsv()
         val packages = listOf(
-            "NPM::glob-parent:5.1.1"
+            "NPM::glob-parent:5.1.1",
+            "NPM::gun:0.7.0" // Package without severity, but with severity inside the databaseSpecific JSON object.
```
[the remark this answers is not in the document]
Comment (fviernau) [reply advisor/src/funTest/kotlin/OsvFunTest.kt:67]: yes.

File: clients/osv/src/main/kotlin/Model.kt (hunk 2 above)
Comment (fviernau) [reply clients/osv/src/main/kotlin/Model.kt:28]: I've seen this comment and tried to write code based on it. I haven't figured it out though. As you say "simply", maybe you know how to use it to reach the goal. In particular, it was unclear what this `polymorphicSerlializer` variable is and how to modify `BaseClass` in that code snippet to make it compile. Do you have a hint?

File: advisor/build.gradle.kts [hunk 5]
```diff
@@ -33,6 +33,7 @@ dependencies {
 
     implementation(libs.cvssCalculator)
     implementation(libs.kotlinxCoroutines)
+    implementation(libs.kotlinxSerialization)
```
Comment (sschuberth) [inline advisor/build.gradle.kts:36]: IMO this change plus the comments in `clients/osv/src/main/kotlin/Model.kt` should go to a separate commit as they fix an issue that was present before the other changes in this commit.

File: clients/osv/src/main/kotlin/Model.kt [hunk 6]
```diff
@@ -33,7 +42,6 @@ import kotlinx.serialization.json.JsonObject
  *
  * For the documentation of all entities and properties please refer to above links.
  */
-
```
Comment (sschuberth) [inline clients/osv/src/main/kotlin/Model.kt:36]: Nit: The removal of this blank line should also go to a separate commit.

File: advisor/build.gradle.kts (hunk 5 above)
Comment (fviernau) [reply advisor/build.gradle.kts:36]: What issue do you think this fixes?

Comment (sschuberth) [reply advisor/build.gradle.kts:36]: Sorry, I was mislead by the earlier discussion of (not) using `kotlinx.serialization` classes in the API. Please ignore this comment of mine.

File: advisor/src/funTest/assets/retrieve-package-findings-expected-result.json [hunk 7]
```diff
@@ -49,5 +49,32 @@
         "severity" : "7.5"
       } ]
     } ]
+  } ],
```
Comment (sschuberth) [inline advisor/src/funTest/assets/retrieve-package-findings-expected-result.json:52]: I'm not sure if this was a misunderstanding of one of my earlier comments, but IMO this commit should be squashed with the next one, so test data and implementation are added in the same commit.

Comment (fviernau) [reply advisor/src/funTest/assets/retrieve-package-findings-expected-result.json:52]: My intention was to make the effect of the parsing obvious in the commits. Your proposal would defeat that idea. But I do actually like it and prefer to not squash. Ok?

Comment (sschuberth) [reply advisor/src/funTest/assets/retrieve-package-findings-expected-result.json:52]: Makes sense, let's keep it as-is.
